### ISSUE TITLE
[D&D 3.5 Sheet] Missing Charactersheet Attribute and/or inconsistent macros

### ISSUE THREAD

**Issue by Ictiv** [Roll20/roll20-character-sheets#10890]
Bit of a brave ping here to the maintainer: @hoyer555 

Recently decided to rerun an old campaign for friends who are new to dungeons and dragons, and ran into a bit of a weird glitch.
Sometimes, players' attack macros would simply not work, or, more precisely after some investigation, one part of the macro would cause a glitch, which ignores the first half of the modifiers, and only includes the last two.

The macros in question, are of the following format:
`1d20cs>@{weapon3critmin} + @{bab}[BAB] + @{epicattackbonus}[Epic AB] + @{weapon3stat}[Ability] + @{size}[size] + @{weapon3enh}[Weapon Enh] + @{weapon3focus}[Weapon Focus] + @{globalAttackBonus}[Global bonus] + ?{Flank |No, 0|Yes, 2}[Flank] + ?{Additional Attack Bonus?|0}[Ad'l Atk Bon]`
The culprit appears to be `@{globalAttackBonus}[Global bonus]` which does not resolve to any number, making the final math resolve to "X+Y+Z++A+B" style format, where the result of `@{globalAttackBonus}[Global bonus]` is supposed to be between the double ++ signs. This results in only A+B being shown in the final attack roll.

Weirdly, this doesn't apply to every macro of every weapon, and which macro it applies to seems random (i.e. within the same sheet, I found weapon macros that contain the `@{globalAttackBonus}[Global bonus]` bit as well as macros that simply lack that variable.

Having taken a look at recent history, this variable appears to have been added just 2 days ago, so the problem seems to be two fold:
1) Whatever script updates the macros in pre-existing character sheets simply skipped over some macros for whatever reason in older sheets, resulting on older sheets having macros that don't reference `globalAttackBonus`, while all new ones appear to do so.
2) None of the sheets, neither old nor new actually appear to have the Attribute `globalAttackBonus` set on them, however after testing, if it's added in manually, the macro works fine.

Not sure what would be the simplest or fastest way to solve this issue, but something needs to be done, as the way I look at it, the recent update might have bricked every 3.5 game's attack macros.

All that being said, hope this proves useful, and thanks a lot for continuing to maintain the 3.5 sheets!

**Comment by hoyer555** [comment]
Hi Ictiv,
thanks for your message, I forgot to initialize the variable "globalAttackBonus", I fix the problem.
you can temporarily fix the problem by putting 0 in the top right field "Global Attack bonus" and "Global Damage bonus".

**Comment by hoyer555** [comment]
Hi Ictiv, my request is merged, I made some tests and from my point of view all macros work fine.
Could you check if your sheet has some problems?

**Comment by Ictiv** [comment]
So to give a full answer, I'll clarify what I could test, as there are 3 scenarios that exist where I imagine the system should handle the initialization.
1) Sheets that existed before the global attack/damage update and weren't touched by me.
2) Sheets that existed before, but were adjusted after the update without touching those fields.
3) Sheets that are newly created.
1 and 3 I could test, however since my current campaign is just getting off the ground, I had already fixed all of the sheets in it by hand, so I have no idea what would have happened in scenario 2.
Sadly, it looks like 3 still doesn't work. I create a new sheet just after seeing this message, and both the PC and NPC version of the global attributes are still empty boxes, and checking in the Attributes & Abilities tab, they also don't seem to show up there. Rolled a few attacks, and the same issue still exists.
As for 1, I loaded up one of my older campaigns, and the sheets there also have the missing variable issue still.

One thing I would recommend, is adding a version file to the repository for now, as at the bottom of the sheet, the version is displayed. While it says the sheet hasn't been updated since 2019 (which was a little suspect for me, but I checked and your repository hardcoded that part) the actual version number seems to be automatically adjusted on the sheet.
If the current version number would be available on the repository, In the future it would be easy to confirm if Roll20 actually pushed the changes through, as right now, the lack of change suggests to me this could be a pipeline issue where your latest update simply didn't go live yet for some reason.

**Comment by Ictiv** [comment]
Actually, adding to that, something strange I haven't encountered yet on Git is definitely going on right now.
At the moment, when visiting the sheet's page (https://github.com/Roll20/roll20-character-sheets/tree/master/D%26D_3-5) it notes at the top that the last update was 10 hours ago, and the details suggest that "charsheet_3-5.html" was adjusted, however, the file list claims that file wasn't changed since 5 days ago.
![image](https://user-images.githubusercontent.com/7266095/183821608-e2969044-cfb5-40db-9af6-b5eb6fa16f38.png)
I'm not certain what could be causing this, but it seems possible that the update is caught up in some backend bottleneck, will try and check back tomorrow, see if it got pushed through.

**Comment by Ictiv** [comment]
Alright, this was an odd little circlestrafe of testing, but I think I just learned something new about how Roll20 works under the hood, and the issue _does_ appear to be fixed, and it was a very specifically me issue that prevented it from showing until a minute ago.
Basically, I was copying over some macros from my old game to my new game yesterday, left the computer on, saw this message in the morning, decided to test things, as I had two games up side by side, why not?
I sort of assumed that the new sheets would be generated on the back end of Roll20, but it appears they are generated locally, because now that I had the bright idea to actually reload the page, newly created character sheets and even the bugged one I created earlier are fixed! Apparently, my browser still had the old sheet code in memory and used that to create faulty sheets.

Ignore the previous 2 reports, I'm guessing the weirdness with the merge request and file update times has to do with how git decided to display the time of the file's alteration vs its addition to the master? Not important.
Sheets do in fact work perfect now, thank you!
